src/replication/status.cpp

```diff
@@ -72,31 +79,42 @@ void from_json(const nlohmann::json &j, ReplicationRoleEntry &p) {
       break;
     }
     case memgraph::replication_coordination_glue::ReplicationRole::REPLICA: {
-      std::string ip_address;
+      std::string address;
       // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
       uint16_t port;
-      j.at(kIpAddress).get_to(ip_address);
+      // In V4 we renamed key `replica_ip_address` to `replica_address`
+      if (version == DurabilityVersion::V4) {
+        j.at(kAddress).get_to(address);
+      } else {
+        j.at(kIpAddress).get_to(address);
+      }
```

Comment:
We shouldn't care about the versions here.
Check what's present in the JSON instead.
```
if (j.contains(kIpAddress)) {
        address = j.at(kIpAddress);
      }
```